src/WireFormat.h

```diff
@@ -1605,6 +1648,182 @@ struct TakeIndexletOwnership {
     } __attribute__((packed));
 };
 
+struct TxParticipant {
+    uint64_t tableId;           // Table Id of the participant object.
+    uint64_t keyHash;           // Key Hash of the participant object.
+    uint64_t rpcId;             // Unique (per transaction) participant id.
+
+    TxParticipant()
+        : tableId()
+        , keyHash()
+        , rpcId()
+    {}
+
+    TxParticipant(uint64_t tableId, uint64_t keyHash, uint64_t rpcId)
+        : tableId(tableId)
+        , keyHash(keyHash)
+        , rpcId(rpcId)
+    {
+    }
+
+    bool operator==(const TxParticipant &other) const {
+        return tableId == other.tableId &&
+               keyHash == other.keyHash &&
+               rpcId == other.rpcId;
+    }
+} __attribute__((packed));
+
+struct TxDecision {
+    static const Opcode opcode = Opcode::TX_DECISION;
+    static const ServiceType service = MASTER_SERVICE;
+
+    enum Decision { COMMIT, ABORT, INVALID };
```

Comment:
Document what these mean (especially INVALID: I'm not sure why that  belongs here). Let's discuss.